Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PingServiceImpl.java

```diff
@@ -52,61 +55,56 @@
 
     private static final String WSDL_SUFFIX = "?wsdl";
     private static final String LOG_WSDL_KEY = "logWsdlPing";
+    private static final String MSG_EXCEPTION_TIMEOUT = "Connection timed out";
+
+    private Set<String> timeoutList = new HashSet<>();
 
     @Override
-    public boolean ping(String url) {
-        InputStream is = null;
-        InputStreamReader isReader = null;
-        BufferedReader in = null;
+    public int ping(String url) {
+        int responseCode = HTTP_CLIENT_TIMEOUT;
+        URL webserviceUrl = null;
 
         try {
-            URL webserviceUrl = new URL(prepUrl(url));
-            HttpsURLConnection.setDefaultHostnameVerifier(getHostNameVerifier());
-            HttpURLConnection con = (HttpURLConnection) webserviceUrl.openConnection();
-
-            is = con.getInputStream();
-            isReader = new InputStreamReader(is);
-
-            in = new BufferedReader(isReader);
-            String inputLine;
-            StringBuilder pingOutput = new StringBuilder();
-
-            while ((inputLine = in.readLine()) != null) {
-                pingOutput.append(inputLine);
+            webserviceUrl = new URL(prepUrl(url));
+            if (!timeoutList.contains(webserviceUrl.getHost())) {
+                HttpsURLConnection.setDefaultHostnameVerifier(getHostNameVerifier());
+                HttpURLConnection con = (HttpURLConnection) webserviceUrl.openConnection();
+                responseCode = con.getResponseCode();
+                logWsdl(readInputStreamFrom(con));
+                con.disconnect();
+            } else {
+                LOG.warn("skip-known-timeout-server-host: {}", url);
```

Comment:
Get rid of  the -'s. Also should this be a warning? It seems desired that we skip it, so why not log this at Debug or Trace level?